x-pack/plugins/alerts/server/task_runner/task_runner.ts

```diff
@@ -313,14 +312,10 @@ export class TaskRunner {
       state: await promiseResult<AlertTaskState, Error>(
         this.validateAndExecuteAlert(services, apiKey, alert)
       ),
-      runAt: asOk(
-        getNextRunAt(
-          new Date(this.taskInstance.startedAt!),
-          // we do not currently have a good way of returning the type
-          // from SavedObjectsClient, and as we currenrtly require a schedule
-          // and we only support `interval`, we can cast this safely
-          alert.schedule
-        )
+      schedule: asOk(
+        // fetch the alert again to ensure we return the correct schedule as it may have
+        // cahnged during the task execution
+        (await alertsClient.get({ id: alertId })).schedule
```

Comment:
Yeah, the main reason why this code is here - is to get the latest schedule after task completed it's execution

Reply:
I will confirm but I think this code gets the schedule before the task starts.

Reply:
> I wonder if this could happen after the executor runs 🤔 Right now, I believe, it will load the latest schedule before the `this.validateAndExecuteAlert(services, apiKey, alert)` promise finishes.

Yeah it does :)
We also added a unit test that ensures that if someone moves it - it will fail because it's returning the wrong schedule.

Reply:
Got it, I missed the `await promiseResult` that was already above. 👍